src/prefect/engine.py

```diff
@@ -254,27 +253,45 @@ async def retrieve_flow_then_begin_flow_run(
         )
         return state
 
+    try:
+        parameters = get_call_parameters(
+            flow.fn, call_args=(), call_kwargs=flow_run.parameters
+        )
```

Comment:
This is why you're never seeing a signature mismatch error. This should probably resolve arguments and keyword arguments using the dedicated function first. Why do we do this call here?

Reply:
I think in general I had a bit of a poor understanding of how things executed. In `Flow.run()` we call `parameters = get_call_parameters` before submitting to `enter_flow_run_engine_from_flow_call`. Then we submit to either `create_and_begin_subflow_run` or `create_then_begin_flow_run`. After that step we call `validate_parameters`. 

The call to `get_call_parameters` to `retrieve_flow_then_begin_flow_run` preserves that order.

Reply:
Removing it does expose the error. I'll get the PR body updated